app-backend/common/src/main/scala/cache/CacheClient.scala

```diff
@@ -65,10 +72,12 @@ class CacheClient(client: => MemcachedClient) extends LazyLogging with RollbarNo
   // Suppress asInstanceOf warning because we can't pattern match on the returned type since it's
   // eliminated by type erasure
   @SuppressWarnings(Array("AsInstanceOf"))
-  def localGetOrElse[CachedType](
+  @tailrec
```

Comment:
How is this not complaining? The function isn't tail-recursive! Is the compiler really smart enough to figure out how to compile it to a tail-recursive function without it actually being tail-recursive?

Reply:
Nevermind, apparently the issue is that I don't understand tail recursion :tada: :burrito: